bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/BusyIndicator.java

```diff
@@ -70,6 +70,41 @@ public static void showWhile(Display display, Runnable runnable) {
 		}
 	}
 
+	/**
+	 * If called from a {@link Display} thread, waits for the given
+	 * <code>Future</code> to complete and provides busy feedback using the
+	 * busy indicator. While waiting for completion, pending UI events are processed
+	 * to prevent UI freeze.
+	 *
+	 * If there is no {@link Display} for the current thread, the method returns
+	 * immediately and no feedback is shown to the user.
+	 *
+	 * In any case, the passed {@link Future} is returned for chaining of further operations.
+	 *
+	 * @param future the {@link Future} for which busy feedback is to be shown.
+	 * @return the {@link Future} for chaining.
+	 * @since 3.127
+	 */
+	public static <F extends Future<T>, T> F showWhile(F future) {
+		Objects.requireNonNull(future);
+		if (!future.isDone()) {
+			Display display = Display.findDisplay(Thread.currentThread());
+			if (display != null) {
+				Integer busyId = setBusyCursor(display);
+				try {
+					while (!future.isDone() && !display.isDisposed()) {
+						if (!display.readAndDispatch()) {
```

Comment:
Can't it happen with this code that: there is no SWT event, so readAndDispatch is stuck, but the future gets completed anyway?
In such case, the expectation would be that even without SWT event, the curser changes back from busy to normal when future completes, while this code gives me the impression that the cursor would remain busy even when future is done.

Reply:
How would readAndDispatch stuck? it is supposed to return if the queue is empty.

Reply:
Indeed, I was mixing with the usual `display.sleep()` that goes together with `readAndDispatch`.
Then this looks to me as it's a potentially infinite loop of the form `while (true)` is there is no event going on, which is likely to consume a lot of CPU in that case.

Reply:
I see `Thread.onSpinWait();` covers the wait as it was earlier discussed with @merks, so let's not bother about it futher.